src/components/visualizations/common/proptypes/execution.js

```diff
@@ -0,0 +1,299 @@
+]);
+
+const measureSortItemPropTypes = {
+    measureSortItem: PropTypes.shape({
+        direction: sortDirectionPropTypes.isRequired,
+        locators: PropTypes.arrayOf(locatorItemPropTypes).isRequired
+    }).isRequired
+};
+
+const sortItemPropTypes = PropTypes.oneOfType([
+    PropTypes.shape(attributeSortItemPropTypes),
+    PropTypes.shape(measureSortItemPropTypes)
+]);
+
+const resultSpecPropTypes = {
+    dimension: PropTypes.arrayOf(PropTypes.shape(dimensionPropTypes)),
+    sorts: PropTypes.arrayOf(sortItemPropTypes)
+};
+
+const measureHeaderItemPropTypes = {
+    measureHeaderItem: PropTypes.shape({
+        uri: PropTypes.string,
+        identifier: PropTypes.string,
+        localIdentifier: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        format: PropTypes.string.isRequired
+    }).isRequired
+};
+
+const totalHeaderItemPropTypes = {
+    totalHeaderItem: PropTypes.shape({
+        name: PropTypes.string.isRequired
+    }).isRequired
+};
+
+const responseMeasureGroupHeaderPropTypes = {
+    measureGroupHeader: PropTypes.shape({
+        items: PropTypes.arrayOf(PropTypes.shape(measureHeaderItemPropTypes)).isRequired,
+        totalItems: PropTypes.arrayOf(totalHeaderItemPropTypes)
+    }).isRequired
+};
+
+const responseAttributeHeaderPropTypes = {
+    attributeHeader: PropTypes.shape({
+        uri: PropTypes.string.isRequired,
+        identifier: PropTypes.string.isRequired,
+        localIdentifier: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        totalItems: PropTypes.arrayOf(PropTypes.shape(totalHeaderItemPropTypes)),
+        formOf: PropTypes.shape({
+            name: PropTypes.string.isRequired,
+            uri: PropTypes.string.isRequired,
+            identifier: PropTypes.string.isRequired
+        }).isRequired
+    }).isRequired
+};
+
+const responseHeaderPropTypes = PropTypes.oneOfType([
+    PropTypes.shape(responseMeasureGroupHeaderPropTypes),
+    PropTypes.shape(responseAttributeHeaderPropTypes)
+]);
+
+const responseDimensionPropTypes = {
+    headers: PropTypes.arrayOf(responseHeaderPropTypes).isRequired
+};
+
+const resultAttributeHeaderItemPropTypes = {
+    attributeHeaderItem: PropTypes.shape({
+        uri: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired
+    }).isRequired
+};
+
+const resultMeasureHeaderItemPropTypes = {
+    measureHeaderItem: PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        order: PropTypes.number.isRequired
+    }).isRequired
+};
+
+const resultTotalHeaderItemPropTypes = {
+    totalHeaderItem: PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        type: PropTypes.string.isRequired
+    }).isRequired
+};
+
+const dataValuePropTypes = PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number
+]);
+
+export const AfmPropTypesShape = PropTypes.shape(afmPropTypes);
+export const ResultSpecPropTypesShape = PropTypes.shape(resultSpecPropTypes);
```

Comment:
Why `PropTypesShape`? `PropTypes` is not enough ?

Reply:
Must be shape